packages/core/src/protocol-helpers.ts

```diff
@@ -240,17 +237,17 @@ export default class ProtocolHelpers {
 
     static readStreamFully(stream: NodeJS.ReadableStream): Promise<Buffer> {
         return new Promise<Buffer>((resolve, reject) => {
-            if (stream) {
+            if (stream != null) {
                 const chunks: Array<unknown> = [];
                 stream.on("data", (data) => chunks.push(data));
                 stream.on("error", reject);
                 stream.on("end", () => {
                     if (
-                        chunks[0] &&
+                        chunks[0] != null &&
                         (chunks[0] instanceof Array || chunks[0] instanceof Buffer || chunks[0] instanceof Uint8Array)
                     ) {
                         resolve(Buffer.concat(chunks as Array<Buffer | Uint8Array>));
-                    } else if (chunks[0] && typeof chunks[0] === "string") {
+                    } else if (chunks[0] != null && typeof chunks[0] === "string") {
                         resolve(Buffer.from(chunks.join()));
                     } else {
                         resolve(Buffer.from(chunks as Array<number>));
```

Comment:
Hmm, I am wondering here: What happens if `chunks[0]` is in fact `undefined`? Would that cause an error in the `else` clause?

Reply:
In general, I think if you introduce a helper variable above (e.g., `const firstChunk = chunks[0];`) and use it in the if checks (e.g., `typeof firstChunk === "string"`), the `chunks[0] != null` can be removed.

Reply:
Doing so would yield something like 

```ts
                    const firstChunk = chunks[0];
                    if (
                        firstChunk instanceof Array ||
                        firstChunk instanceof Buffer ||
                        firstChunk instanceof Uint8Array
                    ) {
                        resolve(Buffer.concat(chunks as Array<Buffer | Uint8Array>));
                    } else if (typeof firstChunk === "string") {
                        resolve(Buffer.from(chunks.join()));
                    } else {
                        resolve(Buffer.from(chunks as Array<number>));
                    }
```

I guess the type safety of the method in general should be revisited in the future, though, using only the first element as an indicator for the type of everything will probably work most of the time, but is not rock solid, I suppose.

Reply:
I think there are definitively many parts that can (and maybe should) be improved.. but I guess for the sake of avoiding issues I kept the general structure...